plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java

```diff
@@ -1015,4 +1025,23 @@ public Map<String, org.opensearch.ingest.Processor.Factory> getProcessors(org.op
             );
         return Collections.unmodifiableMap(processors);
     }
+
+    @Override
+    public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings settings) {
+        List<SystemIndexDescriptor> systemIndexDescriptors = new ArrayList<>();
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_AGENT_INDEX, "ML Commons Agent Index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_CONFIG_INDEX, "ML Commons Configuration Index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_CONNECTOR_INDEX, "ML Commons Connector Index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_CONTROLLER_INDEX, "ML Commons Controller Index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_MODEL_GROUP_INDEX, "ML Commons Model Group Index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_MODEL_INDEX, "ML Commons Model Index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_TASK_INDEX, "ML Commons Task Index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_CONVERSATION_META_INDEX, "ML Commons Conversation Meta Index"));
+        systemIndexDescriptors
+            .add(new SystemIndexDescriptor(ML_CONVERSATION_INTERACTIONS_INDEX, "ML Commons Conversation Interactions Index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_MEMORY_META_INDEX, "ML Commons Memory Meta Index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_MEMORY_MESSAGE_INDEX, "ML Commons Memory Message Index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ML_STOP_WORDS_INDEX, "ML Commons Stop Words Index"));
```

Comment:
should this be supported with a Pattern as well? For e.g. ``.plugins-ml-*``

Reply:
It could make sense in this case since the plugin has many system indices that follow the same naming convention.

My advice to plugins is to use a pattern if they create a variable amount of system indices based on a pattern.

An example of a variable amount of indices is multi-tenancy where each tenant has a .kibana index that follows a pattern `.kibana-{tenantName}`. If the names of all of the indices are known then they should be listed out.